core/src/main/scala/com/lightbend/paradox/template/PageTemplateSystem.scala

```diff
@@ -23,7 +23,7 @@ import org.stringtemplate.v4.{ STErrorListener, STRawGroupDir, ST }
 import collection.concurrent.TrieMap
 
 object CachedTemplates {
-  def apply(dir: File, templateName: String): PageTemplate = {
+  def apply(dir: File, templateName: String = PageTemplate.DefaultName): PageTemplate = {
```

Comment:
By definition, there's no reason to redefine a new argument for `dir: File`, this is by default the folder containing all the defined templates (defined initially in the plugin). On the other hand, `templateName: String` allows to specify directly the template file used inside this folder.

If I take your example, the first call (with `new File("template1")`) will select the default `page.st` template inside the `template1` folder, and the second call inside the `template2`, then I don't think they use the same template if this case happens since they use a different directory.

But indeed, maybe I could check if the template directory specified exists, but I'm not sure it is necessary.